eng/scripts/Get-TypeSpec-Folders.ps1

```diff
@@ -42,7 +42,8 @@ else {
 $typespecFolders = @()
 foreach ($file in $changedFiles) {
   if ($file -match 'specification\/[^\/]*\/') {
-    $typespecFolder = (Get-ChildItem -path $matches[0] tspconfig.yaml -Recurse).Directory.FullName | ForEach-Object {if ($_) { [IO.Path]::GetRelativePath($($pwd.path), $_) }}
+    $typespecFolder = (Get-ChildItem -path $matches[0] tspconfig.yaml -Recurse).Directory.FullName
+      | ForEach-Object {if ($_ -and !(Get-ChildITem -force -path $_ .skip-typespec-validation)) { [IO.Path]::GetRelativePath($($pwd.path), $_) }}
```

Comment:
Move filtering to only apply to `$allChangedFiles`

Reply:
this won't give us the correct result. 
The code on line 45
```
if ($file -match 'specification\/[^\/]*\/') {
    $typespecFolder = (Get-ChildItem -path $matches[0] tspconfig.yaml -Recurse).Directory.FullName
    | ForEach-Object {if ($_) { [IO.Path]::GetRelativePath($($pwd.path), $_) }}
```
will add the folder in again. (because of the logic for "changed file should trigger all folders beyond depth 2 folders"
e.g. changedFiles of specification/cognitiveservices/AnomalyDetector should trigger all folders under cognitiveservices.)

In addition, we also want to separate the $allChangedFiles into two categories, one is the ci run, another is the pr change that has an eng folder or root folder change. there's an edge case of a change in both eng folder and skipped test spec directory